Deprecate the 'Build Workspace Symbols' command when jediEnabled = false

d3r3kk opened the pull request:
Fixes #2267

- new Python Language Server rebuilds symbols in background
- remove registrations/command functionality
- add deprecation popup
- ~cleanup prior deprecated code~

- [x] Title summarizes what is changing
- [x] Includes a [news entry](https://github.com/Microsoft/vscode-python/tree/master/news) file (remember to thank yourself!)
- [x] Unit tests & [code coverage](https://codecov.io/gh/Microsoft/vscode-python) are not adversely affected (within reason)
- [x] Works on all [actively maintained versions of Python](https://devguide.python.org/#status-of-python-branches) (e.g. Python 2.7 & the latest Python 3 release)
- [x] Works on Windows 10, macOS, and Linux (e.g. considered file system case-sensitivity)
- [x] *Conditionally* deprecate the Build Workspace Symbols' command. If `python.jediEnabled: true` then we want to still use the same functionality!

brettcannon commented on src/client/workspaceSymbols/main.ts:
```diff
@@ -39,59 +34,4 @@ export class WorkspaceSymbols implements Disposable {
             });
         }
     }
-    private registerCommands() {
```
We need this for Jedi users, so this code should go back in and then only deprecate when using the language server.

ericsnowcurrently commented on src/client/common/featureDeprecationManager.ts:
```diff
@@ -30,16 +33,39 @@ const deprecatedFeatures: deprecatedFeatureInfo[] = [
 
-export interface IFeatureDeprecationManager extends Disposable {
-    initialize(): void;
+export interface IPopupService {
```
Don't we normally put interfaces in the corresponding `types.ts` file?

ericsnowcurrently commented on src/client/common/featureDeprecationManager.ts:
```diff
@@ -30,16 +33,39 @@ const deprecatedFeatures: deprecatedFeatureInfo[] = [
+        return window.showInformationMessage(message, ...items);
+    }
+}
+@injectable()
```
blank line?

ericsnowcurrently commented on src/client/common/terminal/types.ts:
```diff
@@ -57,3 +57,9 @@ export interface ITerminalActivationCommandProvider {
+
+export const IFeatureDeprecationManager = Symbol('IFeatureDeprecationManager');
+
+export interface IFeatureDeprecationManager extends Disposable {
```
Shouldn't this be in `src/client/common/types.ts`, next to the file where the implementation lives?

d3r3kk commented:
@brettcannon when you get a moment could you please re-review. I've put back the code I removed earlier, and changed my approach to deprecating this command (ie. Only deprecated when Python Language Server is active).

d3r3kk commented:
@brettcannon or @ericsnowcurrently, ready for re-review. 

Note: I found another injectable class already existing that handles the popup functionality, and replaced my PopupService with that.

d3r3kk commented:
Retry CI on macOS - timeout
